apps/communications/contacts/js/views/multiple_select.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+/* global Contacts, contacts, utils */
+
+Contacts.MultipleSelect = (function(){
+  var header = document.getElementById('multiple-select-view-header');
+  var saveButton = document.getElementById('save-button');
+  var container = document.getElementById('multiple-select-container');
+  var template = document.getElementById('contactToImportTemplate');
+  var title = document.getElementById('multiple-select-view-title');
+
+  saveButton.addEventListener('click', importAll);
+  header.addEventListener('action', cancelHandler);
+
+  var contactsToImport = [];
+
+  function cancelHandler() {
+    header.removeEventListener('action', cancelHandler);
+    saveButton.removeEventListener('click', importAll);
+    Contacts.cancel();
+  }
+
+  function importAll() {
+    var numDupsMerged = 0,
+        importedContacts = 0;
+
+    contactsToImport.forEach((contact, index) => {
+      contacts.Matcher.match(contact, 'passive', {
+        onmatch: matches => {
+          var callbacks = {
+            success: () => {
+              numDupsMerged++;
+              _doContinue(index);
+            },
+            error: () => _doContinue(index)
+          };
+          contacts.adaptAndMerge(contact, matches, callbacks);
+        },
+        onmismatch: function() {
+          var saving = navigator.mozContacts.save(contact);
+
+          saving.onsuccess = () => {
+            _doContinue(index, true);
+          };
+
+          saving.onerror = (err) => {
+            console.error(err);
+            _doContinue(index);
+          };
+        }
+      });
+    });
+
+    function _doContinue(index, isContactImported) {
+      if (isContactImported) {
+        importedContacts++;
+      }
+
+      if (contactsToImport.length - 1 <= index) {
+        utils.status.show({
+          id: 'vCardContacts-imported',
+          args: {n: importedContacts}
+        }, {
+          id: 'contactsMerged',
+          args: {numDups: numDupsMerged}
+        });
+
+        window.setTimeout(cancelHandler, 3000); // Allows to see status message.
+      }
+    }
+  }
+
+  /**
+   * Allows to preview which contacts will be imported.
+   * @param {mozContact[]} contacts - Every contact already read from vCard.
+   * @param cursor - Reader object to continue reading from vCard.
+   * @param {string} filename - Name of the vcf file we are reading from.
+   */
+  var addContacts = (contacts, cursor, filename) => {
+    title.textContent = filename;
+    contacts.forEach(_doAddContact);
+
+    cursor.onsuccess = (event) => {
+      var contact = event.target.result;
+      if (contact) {
+        _doAddContact(contact);
+        cursor.continue();
+      }
+    };
+
+    function _doAddContact(contact) {
```

Comment:
can we separate the rendering code in an specific function 'renderContact' to be called by the '_doRenderContact' function?